platform/linux-generic/odp_ipsec.c

```diff
@@ -596,6 +598,42 @@ static int ipsec_in_ah_post(odp_packet_t pkt,
 	return 0;
 }
 
+static void
+ipsec_sa_err_stats_update(odp_ipsec_packet_result_t *result)
+{
+	ipsec_sa_t *sa;
+
+	if (odp_likely(ODP_IPSEC_OK == result->status.error.all))
+		return;
+
+	if (result->status.error.sa_lookup)
+		return;
```

Comment:
There are also other cases where result->sa is not valid. For instance, if packet parsing before SA lookup fails, ipsec_in_single() sets alg error. It would also be more robust in general to check the validity of the SA directly.

Reply:
Agreed.